### DISCUSSION TITLE
local machine's jupyter notebook won't print epochs and losses when training a model via CLI

### DISCUSSION THREAD

**Discussion by Ziad-Assem** [ultralytics/ultralytics#4449] [Q&A]
Training on google colab using cli printed the current epochs, approx time, and losses of each epochs. but when i run the same command on my local machine via jupyter notebook, it just runs and doesn't update me or print anything. how can i print the epochs and details i see when i train using the same cli code on google colab?

**Comment by glenn-jocher** [comment]
@Ziad-Assem hello! It sounds like you're experiencing differences in output verbosity when training YOLOv8 models on your local machine compared to Google Colab. This can sometimes happen due to differences in the environment or the way Jupyter Notebooks handle output streams.

To ensure you're seeing the training progress updates (like epochs, time, and losses) in your Jupyter Notebook, you might want to check the following:

1. Ensure that your Jupyter environment is set up correctly and that it's not suppressing the stdout (standard output) stream.
2. Verify that you're using the correct command and flags for training in the CLI mode. Sometimes, specific flags can control the verbosity of the output.
3. If you're running the training in a separate thread or process, make sure that the output is not being redirected or suppressed.

If the issue persists, you might want to consider running your training directly in a terminal or command prompt to see if the behavior differs from the Jupyter Notebook environment.

For more detailed information on training modes and potential configuration options, you can refer to the training documentation at https://docs.ultralytics.com/modes/train. 📘

Keep in mind that the behavior you're experiencing could be specific to your local setup, so you might need to adjust your Jupyter Notebook settings or the way you're initiating the training process to get the desired output.
